app/test_main.py

```diff
@@ -1 +1,48 @@
-# write your code here
+from unittest import mock
+
+import pytest
+from app.main import can_access_google_page
+
+
+@pytest.fixture()
+def mocked_valid_google_url() -> None:
+    with mock.patch("app.main.valid_google_url") \
+            as mocked_valid_url:
+        yield mocked_valid_url
+
+
+@pytest.fixture()
+def mocked_has_internet_connection() -> None:
+    with mock.patch("app.main.can_access_google_page") \
+            as mocked_has_internet_connection:
+        yield mocked_has_internet_connection
+
+
+def test_can_access_google_page_if_url_false(
+        mocked_valid_google_url: object,
+        mocked_has_internet_connection: object
+) -> None:
+    test_url = "https://test_url"
+    mocked_valid_google_url.return_value = False
+    mocked_has_internet_connection.return_value = False
+    assert can_access_google_page(test_url) == "Not accessible"
+
+
+def test_can_access_google_page_if_url_true(
+        mocked_valid_google_url: object,
+        mocked_has_internet_connection: object
+) -> None:
+    test_url = "https://test_url"
+    mocked_valid_google_url.return_value = True
+    mocked_has_internet_connection.return_value = False
+    assert can_access_google_page(test_url) == "Accessible"
```

Comment:
here u have an valid url but has no internet connection 
![image](https://user-images.githubusercontent.com/103201010/211334679-c14421a4-3af7-4643-a8ef-b05d82dfeb00.png)

Reply:
Also add the test for case with internet connection but without valid url